ctapipe/core/tests/test_tool.py

```diff
@@ -340,3 +342,28 @@ class MyTool(Tool):
         json.dump({"MyTool": {"foo": 5}}, f)
 
     assert run_tool(MyTool(), [f"--config={config}"]) == 2
+
+
+def test_tool_raises():
+    class ToolGood(Tool):
+        name = "Good"
+        description = "This runs without an exception."
+
+        def start(self):
+            print("All good.")
+
+    class ToolBad(Tool):
+        name = "Bad"
+        description = "This tool raises an exception."
+
+        def start(self):
+            raise ValueError("1 does not equal 0.")
+
+    assert run_tool(ToolGood(), raises=True) == 0
+
+    with pytest.raises(ValueError):
+        ret = run_tool(ToolBad(), raises=True)
```

Comment:
Add test that with `raises=False` we get a non-zero exit code for the bad tool

Reply:
done